admin-dev/themes/new-theme/js/pages/product/components/categories/index.js

```diff
@@ -79,16 +83,8 @@ export default class CategoriesManager {
   }
 
   initTree() {
-    const initialElements = {};
-
-    this.categoryTree.querySelectorAll(ProductCategoryMap.treeElement).forEach((treeElement) => {
-      const checkboxInput = treeElement.querySelector(ProductCategoryMap.checkboxInput);
-      const categoryId = this.getIdFromCheckbox(checkboxInput);
-      initialElements[categoryId] = treeElement;
-    });
-
     this.categories.forEach((category) => {
-      const item = this.generateCategoryTree(category, initialElements);
+      const item = this.generateCategoryTree(category);
```

Comment:
```suggestion
      const item = this.generateCategoryTree(category, selectedCategories.contains(category.id));
```
The second parameter will indicate if the checkbox must be pre checked

Reply:
Or this check can be done inside the function as you already did, it doesn't matter But you'll have to either provide the category list or store it in a class field, but that would make the component potentially stateful which can result in unexpected bugs